telegram_bot/test_bot.py

```diff
@@ -0,0 +1,154 @@
+import telebot
+from telebot import types
+import csv
+import os
+
+from urllib3.filepost import writer
+
+API_TOKEN = "1827046423:AAEtXx9tssHYBpsWn1fx11s_CKox538GeqU"
+
+bot = telebot.TeleBot(API_TOKEN)
+
+users = {}
+
+
+
+def is_valid_name_surname(name_surname):
+    return not (" " in name_surname or len(name_surname) < 2)
+
+
+@bot.message_handler(content_types=["text"])
+def start(message):
+    user_id = message.from_user.id
+    if message.text == "Регистрация":
+        # create empty user
+        users[user_id] = {}
+        remove_initial_keyboard(user_id, "Как тебя зовут?")
+        bot.register_next_step_handler(message, get_name)
+    elif message.text == "Создание ToDo":
+        remove_initial_keyboard(user_id, "Введите название задания:")
+        bot.register_next_step_handler(message, to_do)
+    else:
+        render_initial_keyboard(user_id)
+
+
+def get_name(message):
+    user_id = message.from_user.id
+    name = message.text.title()
+
+    if is_valid_name_surname(name):
+        users[user_id]["name"] = name.title()
+        bot.send_message(user_id, "Какая у тебя фамилия?")
+        bot.register_next_step_handler(message, get_surname)
+    else:
+        bot.send_message(user_id, "Введите корректное имя")
+        bot.register_next_step_handler(message, get_name)
+
+
+def get_surname(message):
+    surname = message.text
+    user_id = message.from_user.id
+    if is_valid_name_surname(surname):
+        users[user_id]["surname"] = surname.title()
+        bot.send_message(user_id, "Сколько тебе лет?")
+        bot.register_next_step_handler(message, get_age)
+    else:
+        bot.send_message(user_id, "Введите корректную фамилию")
+        bot.register_next_step_handler(message, get_surname)
+
+
+def get_age(message):
+    csv_dir = os.path.join("test_files", "csv")
+    file_path_1 = os.path.join(csv_dir, "test.csv")
+    age_text = message.text
+    user_id = message.from_user.id
+    if age_text.isdigit():
+        age = int(age_text)
+        if not 10 <= age <= 100:
+            bot.send_message(user_id, "Введите реальный возраст, пожалуйста")
+            bot.register_next_step_handler(message, get_age)
+        else:
+            users[user_id]["age"] = int(age)
+            name = users[user_id]["name"]
+            surname = users[user_id]["surname"]
+            question = f"Тебе {age} лет и тебя зовут {name} {surname}?"
+            render_yes_now_keyboard(user_id, question, "reg")
+            with open(file_path_1, "a") as csv_file:
+                names = ["id", "name", "surname", "age"]
+                writer = csv.DictWriter(csv_file, fieldnames=names)
+                writer.writeheader()
+                writer.writerow({"id": user_id, "name": name, "surname": surname, "age": age})
```

Comment:
```suggestion
            is_first_entity = not os.path.exists(file_path_1)
            with open(file_path_1, "a") as csv_file:
                names = ["id", "name", "surname", "age"]
                writer = csv.DictWriter(csv_file, fieldnames=names)
                if is_first_entity:
                    writer.writeheader()
                writer.writerow({"id": user_id, "name": name, "surname": surname, "age": age})
```

Мы должны записывать заголовок только если в первый раз пишем что-то в файл (то есть, только если до записи его не существовало). В обратном случае, при записи каждого нового пользователя мы будем заново записывать заголовки